com.ibm.wala.core/src/com/ibm/wala/dataflow/IFDS/LocalSummaryEdges.java

```diff
@@ -71,12 +67,12 @@ public LocalSummaryEdges() {
    * @param d2 target dataflow fact
    */
   public void insertSummaryEdge(int s_p, int x, int d1, int d2) {
-    int n = getIndexForEntryExitPair(s_p, x);
-    IBinaryNaturalRelation R = summaries.get(n);
+    long index = LongUtil.pack(s_p, x);
```

Comment:
I wonder why there was the extra level of mapping.  My best guess is that representing summaries with an int instead of a long was a space savings.  However, the number of summaries is just the number of callees, so this shouldn't be a big deal, I would hope.  Plus, this may be computed lazily.